src/complete.cpp

```diff
@@ -226,7 +226,7 @@ completion_t &completion_t::operator=(const completion_t &him) {
 
 bool completion_t::is_naturally_less_than(const completion_t &a, const completion_t &b) {
     //for this to work, stable_sort must be used because results aren't interchangeable
-    if ((a.flags & COMPLETE_DONT_SORT) != 0 && (b.flags & COMPLETE_DONT_SORT) != 0) {
+    if (a.flags & b.flags & COMPLETE_DONT_SORT) {
```

Comment:
> I'm more than happy to include my one-liner from the commit log as a comment atop of that.

Fantastic. Other than that I don't see any problems with this change and will be more than happy to merge it to the 3.0 (`major`) branch. I'm hesitant to merge it to the 2.7 (`master`) branch because we're now within a handful of weeks of making that release. And we want to limit changes to the 2.x branch to bug fixes only.